qemu/tests/nvdimm.py

```diff
@@ -0,0 +1,155 @@
+import logging
+
+from autotest.client.shared import error
+from autotest.client.shared import utils
+
+
+class NvdimmTest(object):
+    """
+    Class for NVDIMM test
+    """
+    def __init__(self, params, env):
+        """
+        Init the default values of NvdimmTest object.
+
+        :param params: A dict containing VM preprocessing parameters.
+        :param env: The environment (a dict-like object).
+        """
+        self.session = None
+        self.params = params
+        self.env = env
+
+    def verify_nvdimm(self, vm):
+        """
+        verify nvdimm in monitor and guest
+
+        :params vm: VM object
+        """
+        dimms_expect = set(["dimm-%s" % dev for dev in self.params.objects("mem_devs")])
+        dimms_monitor = set([info['data']['id'] for info in vm.monitor.info("memory-devices")])
+        if not dimms_expect.issubset(dimms_monitor):
+            invisible_dimms = dimms_expect - dimms_monitor
+            raise error.TestFail("%s dimms are invisible in monitor" % invisible_dimms)
+        check_cmd = "test -b %s" % self.params.get("pmem", "/dev/pmem0")
+        if self.session.cmd_status(check_cmd):
```

Comment:
It's not a good idea to use a global session for the entire case without checking it's alive or not.